src/Avalonia.Visuals/Media/DrawingContext.cs

```diff
@@ -119,16 +119,40 @@ public void DrawGeometry(IBrush brush, IPen pen, Geometry geometry)
         }
 
         /// <summary>
-        /// Draws the outline of a rectangle.
+        /// Draws a rectangle with the specified Brush and Pen. The pen and the brush can be null.
         /// </summary>
+        /// <param name="brush">The brush.</param>
         /// <param name="pen">The pen.</param>
         /// <param name="rect">The rectangle bounds.</param>
-        /// <param name="cornerRadius">The corner radius.</param>
-        public void DrawRectangle(IPen pen, Rect rect, float cornerRadius = 0.0f)
+        /// <remarks>
+        /// The brush and the pen can both be null. If the brush is null, then no fill is performed.
+        /// If the pen is null, then no stoke is performed. If both the pen and the brush are null, then the drawing is not visible.
+        /// </remarks> 
+        public void DrawRectangle(IBrush brush, IPen pen, Rect rect)
         {
-            if (PenIsVisible(pen))
+            if (brush != null || PenIsVisible(pen))
+            {
+                PlatformImpl.DrawRectangle(brush, pen, rect);
+            }
+        }
+
+        /// <summary>
+        /// Draws a rounded rectangle with the specified Brush and Pen.
+        /// </summary>
+        /// <param name="brush">The brush.</param>
+        /// <param name="pen">The pen.</param>
+        /// <param name="rect">The rectangle bounds.</param>
+        /// <param name="radiusX"></param>
+        /// <param name="radiusY"></param>
+        /// <remarks>
+        /// The brush and the pen can both be null. If the brush is null, then no fill is performed.
+        /// If the pen is null, then no stoke is performed. If both the pen and the brush are null, then the drawing is not visible.
+        /// </remarks>
+        public void DrawRoundedRectangle(IBrush brush, IPen pen, Rect rect, double radiusX, double radiusY)
```

Comment:
Why do we need a separate method instead of default argument value as we did before?

Reply:
This just matches the Wpf API. We can use a single method with default parameters if you want.

Reply:
I'm undecided on this. I guess WPF compatibility is a useful thing to have, but how compatible are we on this API of this class in general?

Reply:
DrawRectangle now has the same signature just a different name.